api/pkg/apis/v1alpha1/managers/solution/solution-manager.go

```diff
@@ -64,6 +64,7 @@ type SolutionManager struct {
 	IsTarget        bool
 	TargetNames     []string
 	ApiClientHttp   api_utils.ApiClient
+	jobList         map[string]map[string]map[string]context.CancelFunc
```

Comment:
You need some way to protect the map. Map is not thread-safe in golang. Considering to use https://pkg.go.dev/sync#Map

Reply:
Agreed. And this map of map of map looks scary. Maybe we should create a struct for the type?

Reply:
I created a struct for {namespace, name}. For jobId, since we need to iterate it when cancel previous jobs, I cannot find a good way to pair it together with namespace and name. The drawback of this is I cannot use sync.Map (because it would be inconvenient to handle internal map in an atomic operation)